_rules/image-no-text-0va7u6.md

```diff
@@ -0,0 +1,205 @@
+---
+id: 0va7u6
+name: HTML graphics contain no text
+rule_type: atomic
+description: |
+  This rule checks that images of text are not used
+accessibility_requirements:
+  wcag20:1.4.5: # Images of Text (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag20:1.4.9: # Images of Text (No Exception) (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Carlos Duarte
+  images:
+    - Letter posted from Dulverton (Somerset, England) to Bristol (England) in 1894. Released into the public domain by Adrian Pingstone.
+    - Times Square image released into the public domain by (WT-shared) Ypsilonatshared at wts wikivoyage.
+---
+
+## Applicability
+
+This rule applies to any [visible][] [embedded image][] that is not [marked as decorative][].
+
+## Expectation
+
+For each of the image resources referenced by the [image sources][] of the test target, at least one of the following is true:
+
+- the image resource does not contain text expressing anything in a [human language][]; or
+- the image resource contains text and the text is not the most significant visual content in the image; or
+- the image resource contains text and it is [essential][] that the text is rendered with that specific presentation.
```

Comment:
I think we need to add the following situations as well:

- the image contains text, and the font, size, color, and background of the text in the image can be set;
- the image contains text, and the visual presentation of the text in the image cannot be achieved through CSS

Reply:
Regarding the first point: as far as I know, all the options in the "embedded image" definition do not allow setting font, size, color and background. Note that SVGs are only applicable if the image is embedded through the `image` tag. And SVGs being used as source for the other elements do not allow that as Wilco pointed out (and I confirmed).

Regarding the second point: what is the difference between that and the last point? If I cannot achieve the presentation with CSS and I chose to use an image of text then it must be because rendering the text with that specific presentation is essential. Otherwise, I wouldn't use an image of text and would render the text with a different presentation.

Reply:
Regarding the first point, I think there are situations when you can set the font, size, color, etc. of text that is generated inside embedded images -- for example in some meme generators. The image itself is static but the script/app can regenerate the image according to some settings/preferences.

Regarding the second point, I think "essential" is like a logo or such -- even if that presentation can be done by CSS, it may be deemed necessary to keep the exact look or proportions, etc. I think 1.4.5 allows also non-essential text to be presented via images if that styling cannot be done by CSS (these situations have become rare with the steady improvements to CSS).

Reply:
I've been trying to find a meme generator that allows for such customisation (in run-time, not in generation-time) but I couldn't find one. I would really need to have a look at one so that I can understand if what is being used is something that falls under our definition of embedded image. 

Regarding that second point, what you're asking is a huge problem. Adding such a clause (it passes if there is no way to achieve that presentation via css - or other technology) would make it the burden of the evaluator to prove that there is no way to achieve such presentation with css. There would be situations where an evaluator would not be able to demonstrate that. But that does not mean it isn't possible, just that the evaluator does not know how to achieve it. So, we can't have that has an expectation. Would an assumption such as "the rule assumes that for the images of text it is not possible to achieve the desired presentation with text-based technology" work for you?

Reply:
> I've been trying to find a meme generator that allows for such customisation (in run-time, not in generation-time) but I couldn't find one. I would really need to have a look at one so that I can understand if what is being used is something that falls under our definition of embedded image.

I guess we can definitely imagine a page with an image of text and some scripting that allow resizing that text (e.g. trough a slider), maybe by having a client-side mash-up of the image + text or something. As often, there isn't anything we can't do with JS 😕 

In that case, the meme generator page itself is good (allow resize), but the result is bad (does not allow resize).

But that could probably go through an assumption that the page does not provide ways of resizing the text. Assumption because anything that is dynamic (JS) yield undecidable problems as soon as we try to look into it, so it can't really be in expectations.

Reply:
OK. Included two new assumptions to account for these two cases.

Reply:
I don't have sufficient knowledge to assess the inaccuracy that this approach introduces but I feel quite ambivalent about it. This is now a rule for 1.4.9 rather than for 1.4.5. Happy to discuss.

Reply:
I believe the second condition in the expectation makes this a rule for 1.4.5.
Why do you say that this is only about 1.4.9?